File: javascript/bubble_sort.js
function generate_sample_array(size = 30) {
  return Array.from({ length: size }).map(() =>
    Math.floor(Math.random() * 100),
  );
}

export default function bubble_sort(arr) {
  for (let i = 0; i < arr.length; i++) {
    for (let j = 0; j < arr.length - 1 - i; j++) {
      if (arr[j] > arr[j + 1]) {
        const tmp = arr[j];
        arr[j] = arr[j + 1];
        arr[j + 1] = tmp;
      }
    }
  }
}

const sample = generate_sample_array();
console.log("Before sort:", sample);
bubble_sort(sample);
console.log("After sort:", sample);

File: lua/bubble_sort.lua
local input = { 8, 20, 5, 1, 9, 47, 82, 69, 124, 2 }

local bubble_sort = function(list)
  for i = 1, #list do
    for j = 1, #list - i do
      if list[j] > list[j + 1] then
        list[j], list[j + 1] = list[j + 1], list[j]
      end
    end
  end
end

print("Before:", table.concat(input, ','))
bubble_sort(input)
print("After:", table.concat(input, ','))
